### PR TITLE
feat: use real RA-TLS for everything

### PR DESCRIPTION
**Pull request by haraldh**
* add `tee-ratls-preexec` for creating the vault certificate
* remove the old attestation API

### CODE REVIEW

File: src/client/mod.rs [hunk 1]
```diff
@@ -246,6 +121,8 @@ impl TeeConnection {
                 _ocsp_response: &[u8],
                 _now: UnixTime,
             ) -> Result<rustls::client::danger::ServerCertVerified, rustls::Error> {
+                // TODO: DO RA-TLS verification here
```
Comment (thomasknauth) [inline src/client/mod.rs:124]: Looks like you implemented the verification? Can the TODO be removed?

File: src/client/mod.rs [hunk 2]
```diff
@@ -255,20 +132,134 @@ impl TeeConnection {
...
+                    collateral_expired,
+                    result,
+                    quote,
+                    advisories,
+                    earliest_expiration_date,
+                    ..
+                } = verify_quote_with_collateral(quote_bytes, collateral.as_ref(), current_time)
+                    .unwrap();
+
+                if collateral_expired || !matches!(result, sgx_ql_qv_result_t::SGX_QL_QV_RESULT_OK)
```
Comment (thomasknauth) [inline src/client/mod.rs:200]: Maybe Rust n00b, but why not `result !=  sgx_ql_qv_result_t::SGX_QL_QV_RESULT_OK`?

File: src/client/mod.rs [hunk 3]
```diff
@@ -255,20 +132,134 @@ impl TeeConnection {
...
+                            "mrenclave mismatch: got {}, expected {}",
+                            hex::encode(quote.report_body.mrenclave),
+                            &mrenclave
+                        )));
+                    } else {
+                        info!("mrenclave `{mrenclave}` matches");
+                    }
+                }
+
+                Ok(rustls::client::danger::ServerCertVerified::assertion())
```
Comment (thomasknauth) [inline src/client/mod.rs:262]: Is this just a weird API? Calling `assertion()` here feels wrong. Looking at the API, I guess they want to make it stick out.

File: src/server/pki.rs [hunk 4]
```diff
@@ -3,213 +3,281 @@
 
 //! Some cryptographic utilities
```
Comment (thomasknauth) [inline src/server/pki.rs:4]: I guess a more meaningful summary might be in order, maybe "Routines for en-/decode custom X509 extensions"?

File: src/client/mod.rs (hunk 1 above)
Comment (haraldh) [reply src/client/mod.rs:124]: right :)

File: src/client/mod.rs (hunk 3 above)
Comment (haraldh) [reply src/client/mod.rs:262]: yeah, it is weird

File: src/client/mod.rs (hunk 2 above)
Comment (haraldh) [reply src/client/mod.rs:200]: Thanks! Must be some leftover.

File: src/client/mod.rs (hunk 1 above)
Comment (haraldh) [reply src/client/mod.rs]: removed

File: src/client/mod.rs (hunk 2 above)
Comment (haraldh) [reply src/client/mod.rs]: fixed

File: src/server/pki.rs (hunk 4 above)
Comment (haraldh) [reply src/server/pki.rs]: reworded

**Review by thomasknauth — APPROVED** [review]

**Pull request merged**